extension/source/contentScript.ts

```diff
@@ -0,0 +1,139 @@
+/**
+ * The only thing this does is inject the in-page script and relay communication
+ * between that in-page script and the background script.
+ *
+ * Why do we need to do it that way? That's a good question.
+ *
+ * The in-page script cannot talk directly to the background script. This is
+ * not our design, it's a limitation of how WebExtensions work which is
+ * motivated by security. The content script is trustable because it's an
+ * independent JavaScript context. This means the dApp cannot interfere with
+ * what the content script does. E.g. If the dApp does any monkeypatching or
+ * prototype pollution, the content script will not be affected.
+ *
+ * We do make use of this trusted context feature by setting the origin and
+ * providerId here, so that the background script can safely decide which
+ * information is appropriate for its responses using these fields.
+ *
+ * However, the browser could easily provide this kind of information securely
+ * as part of an api that allows direct communication between the in-page script
+ * and the background script. If that ever happens, we should most likely use
+ * that and delete the content script entirely.
```

Comment:
https://github.com/web3well/bls-wallet/issues/171 -> https://developer.chrome.com/docs/extensions/mv3/migrating_to_service_workers/

Based on understanding of that last link I think we will need to do something like this when we move to v3. There also has been some talk on using an extension registry instead of `window.ethereum` that would communicate either with `iframe`s/`postMessage` or WebRTC, the later of which would just be like connecting to a socket as a message bus.